### PR TITLE
reporter: Show the repository configuration in the static HTML report

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Show the used repository configuration at the bottom of the static HTML
report. Use https://prismjs.com for syntax highlighting, which is
published under the MIT license.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/1412)

### CODE REVIEW

File: reporter/src/main/kotlin/reporters/StaticHtmlReporter.kt [hunk 1]
```diff
@@ -328,8 +330,19 @@ class StaticHtmlReporter : Reporter() {
                         +css
                     }
                 }
+                link(
+                    rel = "stylesheet",
+                    href = "//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/default.min.css"
+                )
+                script(src = "//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/highlight.min.js") {}
```
Comment (sschuberth) [inline reporter/src/main/kotlin/reporters/StaticHtmlReporter.kt]: `https:` is missing here. Also, are the empty curly braces really required?

File: reporter/src/main/kotlin/reporters/StaticHtmlReporter.kt [hunk 2]
```diff
@@ -328,8 +330,19 @@ class StaticHtmlReporter : Reporter() {
                         +css
                     }
                 }
+                link(
+                    rel = "stylesheet",
+                    href = "//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/default.min.css"
```
Comment (sschuberth) [inline reporter/src/main/kotlin/reporters/StaticHtmlReporter.kt]: `https:` is missing here.

File: reporter/src/main/kotlin/reporters/StaticHtmlReporter.kt (hunk 1 above)
Comment (mnonnenmacher) [reply reporter/src/main/kotlin/reporters/StaticHtmlReporter.kt]: I took this from the docs, but it turns out the protocol is not added when loading the report from a local file, so added `https:`.
Yes, the braces are required, there is no default value for the parameter.

File: reporter/src/main/kotlin/reporters/StaticHtmlReporter.kt (hunk 2 above)
Comment (mnonnenmacher) [reply reporter/src/main/kotlin/reporters/StaticHtmlReporter.kt]: Done.

**Comment by sschuberth** [comment]
I just checked the YAML syntax highlighting in the generated HTML:

image

That's not actually very much colored, with only two colors. Also, "excludes" is red for some reason, which might even make sense here, but is a coincidence, I guess. Long story short, does syntax highlighting really add value here, or should we drop it for simplicity? Also, previously the report was usable offline, but now it requires an online connection for `highlight.js`.

**Comment by mnonnenmacher** [comment]
> That's not actually very much colored, with only two colors.

How many colors would you expect for YAML? :)

> Also, "excludes" is red for some reason, which might even make sense here, but is a coincidence, I guess.

I think this was because the default build of highlight.js does not include YAML, so some other language was auto-detected.

> Long story short, does syntax highlighting really add value here, or should we drop it for simplicity?

I think the integration is still simple and it's even though it's only two colors it still looks much better than black on white.

> Also, previously the report was usable offline, but now it requires an online connection for `highlight.js`.

I have embedded this now to make it work offline, and switched to PrismJS because it is a bit smaller.

**Pull request merged**